9 service authentication

AndHager opened the pull request:
Implementation of the AuthGuard, AuthService and the AuthenticationComponent
Closes: #9

AndHager committed:
Implemented authentication

AndHager committed:
Implemented AuthGuard, AuthService and AuthenticationComponent

AndHager committed:
Merge branch 'develop' of https://github.com/electron2302/continuous-time-tracking into 9-service-authentication

AndHager committed:
Added back navigation

walli545 commented on ctt-frontend/src/app/services/auth.service.ts:
```diff
@@ -0,0 +1,47 @@
+  usernameSubject = new Subject<string>();
+  isSignedIn = false;
+  title = 'ctt-frontend';
+  user: CognitoUserInterface | undefined;
```
Please make all properties private and provide `isSignedIn` via a method => Encapsulation
Also remove unused `title` as well.
`user` and `authState` can be reduced to local scope.

walli545 commented on ctt-frontend/src/app/services/auth.service.ts:
```diff
@@ -0,0 +1,47 @@
+    return this.signedInSubject.asObservable();
+  }
+
+  username(): Observable<String> {
```
```suggestion
  usernameObservable(): Observable<String> {
```

electron2302 commented:
Pleas merge develop to add CI

Commit:
Update ctt-frontend/src/app/services/auth.service.ts

Co-authored-by: walli545 <walli5446@googlemail.com>

AndHager committed:
Applied sugested changes

AndHager committed:
Merge branch '9-service-authentication' of https://github.com/electron2302/continuous-time-tracking into 9-service-authentication

AndHager committed:
Removed unsued title

AndHager committed:
Merge branch 'develop' of https://github.com/electron2302/continuous-time-tracking into 9-service-authentication

AndHager committed:
Implemented log-out redirection

AndHager committed:
Merge branch 'develop' of https://github.com/electron2302/continuous-time-tracking into 9-service-authentication

walli545 committed:
Fix auth routing

Components where appended when routing to /login. THis happend because the this.router.navigateByUrl call happens in the wrong zone. Fixed by using NgZone.

walli545 committed:
Move auth guard to guards/

walli545 committed:
Merge branch '9-service-authentication' of github.com:electron2302/continuous-time-tracking into 9-service-authentication

AndHager committed:
removed tests

AndHager committed:
removed tests
